simple/provider.go

```diff
@@ -53,15 +81,17 @@ func (p *Provider) Provide(root cid.Cid) error {
 
 // Handle all outgoing cids by providing (announcing) them
 func (p *Provider) handleAnnouncements() {
-	for workers := 0; workers < provideOutgoingWorkerLimit; workers++ {
+	for workers := 0; workers < p.workerLimit; workers++ {
 		go func() {
 			for p.ctx.Err() == nil {
 				select {
 				case <-p.ctx.Done():
 					return
 				case c := <-p.queue.Dequeue():
+					ctx, cancel := context.WithTimeout(p.ctx, p.timeout)
```

Comment:
Right now default timeout is 0 and the context will be cancelled immediately.

Reply:
I suggest changing the `mockRouting` in the tests to fail if the context is expired.

Reply:
Yeah, I'll add a few tests around timing out.